File: sort-algorithms/merge-sort.js
function mergeSort(arr) {
  if (arr.length > 1) {
    const div = Math.ceil(arr.length / 2);
    let left = arr.slice(0, div);
    let right = arr.slice(div);

    mergeSort(left);
    mergeSort(right);

    let i = 0,
      j = 0,
      k = 0;

    while (i < left.length && j < right.length) {
      if (left[i] < right[j]) {
        arr[k++] = left[i++];
      } else {
        arr[k++] = right[j++];
      }
    }

    while (i < left.length) arr[k++] = left[i++];

    while (j < right.length) arr[k++] = right[j++];
  }

  console.log(arr);
  return arr;
}

mergeSort([8, 5, 1, 4, 2, 3]);

File: sort-algorithms/merge_sort.py
import numpy as np

def merge_sort(arr):
    if len(arr) > 1:
        div = len(arr)//2
        left = arr[:div].copy()
        right = arr[div:].copy()

        merge_sort(left)
        merge_sort(right)

        i = j = k = 0

        while i < len(left) and j < len(right):
            if left[i] < right[j]:
                arr[k] = left[i]
                i += 1
            else:
                arr[k] = right[j]
                j += 1
            k += 1
        
        while i < len(left):
            arr[k] = left[i]
            i += 1
            k += 1

        while j < len(right):
            arr[k] = right[j]
            j+=1
            k +=1

    return arr

merge_sort(np.array([8,5,1,4,2,3], dtype=int))